src/dataplane-protocol/src/smartstream.rs

```diff
@@ -24,6 +36,36 @@ mod encoding {
         pub error: Option<SmartStreamRuntimeError>,
     }
 
+    /// Indicates an internal error from within a SmartStream.
+    //
+    // The presence of one of these errors most likely indicates a logic bug.
+    // This error type is `#[repr(i32)]` because these errors are returned
+    // as the raw return type of a Smartstream WASM function, i.e. the return
+    // type in `extern "C" fn filter(ptr, len) -> i32`. Positive return values
+    // indicate the numbers of records, and negative values indicate various
+    // types of errors.
+    //
+    // THEREFORE, THE DISCRIMINANTS FOR ALL VARIANTS ON THIS TYPE MUST BE NEGATIVE
+    #[repr(i32)]
+    #[derive(thiserror::Error, Debug, Clone, PartialEq, Encoder, Decoder)]
+    #[fluvio(encode_discriminant)]
+    pub enum SmartStreamInternalError {
+        #[error("encountered unknown error during Smartstream processing")]
+        UnknownError = -1,
```

Comment:
is there any reason why this is negative?

Reply:
I commented why right above on this struct 😂 

```
    // The presence of one of these errors most likely indicates a logic bug.
    // This error type is `#[repr(i32)]` because these errors are returned
    // as the raw return type of a Smartstream WASM function, i.e. the return
    // type in `extern "C" fn filter(ptr, len) -> i32`. Positive return values
    // indicate the numbers of records, and negative values indicate various
    // types of errors.
    //
    // THEREFORE, THE DISCRIMINANTS FOR ALL VARIANTS ON THIS TYPE MUST BE NEGATIVE
```

Reply:
Got it.   Sorry didn't see that